pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -597,3 +601,43 @@ func componentFromChartName(chartName string) string {
 	_, componentName := path.Split(chartName)
 	return componentName
 }
+
+func overrideDiscoverySelectors(spec *v2.ControlPlaneSpec, controlPlaneNamespace string) {
+	var discoverySelectors []*metav1.LabelSelector
+	if spec.MeshConfig != nil {
+		discoverySelectors = spec.MeshConfig.DiscoverySelectors
+	}
+	if discoverySelectors == nil {
+		return
+	}
+	for _, s := range discoverySelectors {
+		if s.MatchExpressions != nil {
+			for _, e := range s.MatchExpressions {
+				if e.Key == "kubernetes.io/metadata.name" {
+					if sliceContains(e.Values, controlPlaneNamespace) {
+						return
+					}
+				}
+			}
+		}
+	}
+	discoverySelectors = append(discoverySelectors, &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "kubernetes.io/metadata.name",
+				Operator: metav1.LabelSelectorOpIn,
+				Values:   []string{controlPlaneNamespace},
+			},
+		},
+	})
```

Comment:
Hmm, aren't discovery selectors ANDed together (and not ORed)? This means that by adding this selector, discovery will _only_ be performed in the control plane namespace.

Also, what if there's an existing `kubernetes.io/metadata.name` selector? In that case, you have to append to it.

Reply:
Ah, nevermind. We have an array of selectors that are ORed, but each selectors contains an array of requirements that are ANDed together, so your solution should be okay.